nim_status/lib/settings/types.nim

```diff
@@ -0,0 +1,141 @@
+
+import options, json, json_serialization
+import  json_serialization/[reader, writer, lexer]
+import utils
+import web3/[conversions, ethtypes]
+
+
+type
+  SettingsError* = object of CatchableError
+
+  SettingsType* {.pure.} = enum
+    Address = "address",
+    ChaosMode = "chaos-mode?",
+    Currency = "currency",
+    CustomBootNodes = "custom-bootnodes",
+    CustomBootNodesEnabled = "custom-bootnodes-enabled?",
+    DappsAddress = "dapps-address",
+    EIP1581Address = "eip1581-address",
+    Fleet = "fleet",
+    HideHomeToolTip = "hide-home-tooltip?",
+    InstallationId = "installation-id",
+    KeyUID = "key-uid",
+    KeyCard_InstanceUID = "keycard-instance_uid",
+    Keycard_PairedOn = "keycard-paired_on",
+    Keycard_Pairing = "keycard-pairing",
+    LastUpdated = "last-updated",
+    LatestDerivedPath = "latest-derived-path",
+    LogLevel = "log-level",
+    Mnemonic = "mnemonic",
+    Name = "name",
+    CurrentNetwork = "networks/current-network",
+    Networks = "networks/networks",
+    NodeConfig = "node-config",
+    NotificationsEnabled = "notifications-enabled?",
+    PhotoPath = "photo-path",
+    PinnedMailservers = "pinned-mailservers",
+    PreferredName = "preferred-name",
+    PreviewPrivacy = "preview-privacy?",
+    PublicKey = "public-key",
+    RememberSyncingChoice = "remember-syncing-choice?",
+    RemotePushNotificationsEnabled = "remote-push-notifications-enabled?",
+    PushNotificationsServerEnabled = "push-notifications-server-enabled?",
+    PushNotificationsFromContactsOnly = "push-notifications-from-contacts-only?",
+    PushNotificationsBlockMentions = "push-notifications-block-mentions?",
+    SendPushNotifications = "send-push-notifications?",
+    SigningPhrase = "signing-phrase",
+    StickersPacksInstalled = "stickers/packs-installed",
+    StickersPacksPending = "stickers/packs-pending",
+    StickersRecentStickers = "stickers/recent-stickers",
+    SyncingOnMobileNetwork = "syncing-on-mobile-network?",
+    UseMailservers = "use-mailservers?",
+    Usernames = "usernames",
+    WalletRootAddress = "wallet-root-address",
+    WalletSetUpPassed = "wallet-set-up-passed?",
+    WalletVisibleTokens = "wallet/visible-tokens",
+    Appearance = "appearance",
+    WakuEnabled = "waku-enabled"
+    WakuBloomFilterMode = "waku-bloom-filter-mode",
+    WebviewAllowPermissionRequests = "webview-allow-permission-requests?"
```

Comment:
This is the same as the enum with have in `nim-status-client` right? I guess once this is merged, we could use this `enum` instead of the one in the client repo

Reply:
Yeah!